### DISCUSSION TITLE
Issue resolving imports and artifacts since today... anyone else?

### DISCUSSION THREAD

**Discussion by abenz1267** [mfussenegger/nvim-jdtls#667] [General]
Hello,

i'm not sure if that's only me and my system rn, but i'm getting errors with some imports and artifacts today.

- VSCode works fine
- building locally via maven works fine as well

It's definitely not a problem with the project itself. I can't figure out why i'm getting those errors out of a sudden.

Anyone else? Stable nvim, plugins all up-to-date.

Regards

**Comment by abenz1267** [comment]
This doesn't happen on my second machine, same config.

**Reply by vishal423** [reply]
Could you try `:JdtWipeDataAndRestart` to clean workspace and see if the error persists?

**Reply by abenz1267** [reply]
... i mean ... that's the first thing i did ofc. Even re-installed nvim being paranoid there might some some weird corruption going on. No success!

Only different between the machine where i get the error and the one where it works fine is the hardware. If i can't fix it next monday (... only in office on mondays...), i'll just re-install arch and hope that fixes it.

But i didn't do anything, besides updating arch and updating nvim ( and plugins ).

**Reply by vishal423** [reply]
I still suspect the issue is with your project workspace. Could you try manually deleting the project directory from your workspace location and then open project in `nvim`? BTW, do you see any errors on `jdtls` loading?

**Reply by abenz1267** [reply]
i already did that. no, no errors when jdtls loads.

**Reply by vishal423** [reply]
I see. I assume you are using correct jdt runtime. If you could share error screenshots related to imports, then, it may help diagnose the issue?

**Reply by abenz1267** [reply]
Sorry, i'm at the office again next monday. But it's just generic "unresolved import"-kinda errors. And maven artifact errors. But yeah... it builds and runs fine on VSCode on that machine. On my current machine: no problem at all. I'm lost. :D

**Reply by abenz1267** [reply]
Finally back at work. So what i did now:

- updated my system (Arch)
- removed `.m2` folder
- removed `workspace` folder
- reinstalled all LSPs and nvim plugins
- did a maven clean

Errors are still present: `The import *redacted* cannot be resolved`. The file is literally present in the project though.

I don't understand why I'm getting those issues on this machine, but not on my other one. Same configuration.

Edit: When jdtls was initializing, it was missing a lot of artifacts... but running mvn clean should have installed all those, or not?

**Reply by abenz1267** [reply]
It complains about missing artifacts that are definitely present in my `.m2` folder. Otherwise building with maven from cmd-line wouldn't work...

**Reply by abenz1267** [reply]
Ok, i've fixed the issue by copying my `/usr/share/java/maven/conf/settings.xml` into `~/.m2`.
